### ISSUE TITLE
max_threads to execute query

### ISSUE THREAD

**Issue by chenziliang** [timeplus-io/proton#569]
**Describe the situation**

The current [max_threads](https://github.com/timeplus-io/proton/blob/develop/src/QueryPipeline/QueryPipelineBuilder.h#L153-L158) which is used to execute a streaming query is not optimal and quite lots of times a waste of lots idle threads. The max threads required for a streaming query is tired to number of shards / partitions (united) of the query. 

Please note, we need be careful to calculate the max_threads to at least as big as total shards, otherwise for join etc, the query may be not able to progress. 

**How to reproduce**
* Which Proton server version to use
* Which interface to use, if matters
* Non-default settings, if any
* `CREATE TABLE` statements for all tables involved
* Queries to run that lead to slow performance

**Expected performance**

**Additional context**

**Comment by marvin-hansen** [comment]
You can customize the number of threads with a custom configuration file. Follow these instructions. 

https://github.com/timeplus-io/proton/wiki/Proton-Server-Setup-Guide(with-configuration-guide)#switching-to-config-file-for-detailed-control-part-4

**Comment by chenziliang** [comment]
Yes,and the default setting shall be optimal. And we are fixing the default max_threads

**Comment by yl-lisen** [comment]
better default threads to executing streaming query in 3.0
